examples/rust-sdk/lp-bot/src/utils.rs

```diff
@@ -0,0 +1,329 @@
+            let signature = rpc.send_and_confirm_transaction(&transaction).await?;
+            Ok(signature)
+        },
+        MAX_RETRIES,
+        INITIAL_RETRY_DELAY,
+        "send transaction",
+    )
+    .await
+}
+
+async fn get_priority_fee_instruction(
+    rpc: &RpcClient,
+    instructions: &[solana_sdk::instruction::Instruction],
+    wallet: &dyn Signer,
+    additional_signers: &[&dyn Signer],
+    tier: PriorityFeeTier,
+    max_priority_fee_lamports: u64,
+) -> Result<Option<solana_sdk::instruction::Instruction>, Box<dyn std::error::Error>> {
+    if let Some(priority_fee_micro_lamports) = calculate_priority_fee(rpc, tier).await? {
+        let recent_blockhash = rpc.get_latest_blockhash().await?;
+        let message = Message::new(instructions, Some(&wallet.pubkey()));
+        let mut signers = vec![wallet];
+        signers.extend(additional_signers);
+
+        let transaction = Transaction::new(&signers, message, recent_blockhash);
+        let simulated_transaction = rpc.simulate_transaction(&transaction).await.unwrap();
+
+        if let Some(units_consumed) = simulated_transaction.value.units_consumed {
+            let mut compute_unit_price = priority_fee_micro_lamports;
+            let total_priority_fee_lamports =
+                (units_consumed as u64 * priority_fee_micro_lamports) / 1_000_000;
+
+            if total_priority_fee_lamports > max_priority_fee_lamports {
+                compute_unit_price = (max_priority_fee_lamports * 1_000_000) / units_consumed;
+            }
+
+            display_priority_fee_details(
+                compute_unit_price,
+                units_consumed,
+                (units_consumed as u64 * compute_unit_price) / 1_000_000,
+            );
+
+            return Ok(Some(create_priority_fee_instruction(compute_unit_price)));
+        }
+    }
+
+    Ok(None)
+}
+
+fn display_priority_fee_details(
+    compute_unit_price: u64,
+    units_consumed: u64,
+    total_priority_fee_lamports: u64,
+) {
+    println!(
+        "Priority Fee Details:\n\
+        - Compute Unit Price: {} microlamports\n\
+        - Compute Units Consumed: {}\n\
+        - Total Priority Fee: {} lamports",
+        compute_unit_price, units_consumed, total_priority_fee_lamports
+    );
+}
+
+async fn calculate_priority_fee(
+    rpc: &RpcClient,
+    tier: PriorityFeeTier,
+) -> Result<Option<u64>, Box<dyn std::error::Error>> {
+    let prioritization_fees = rpc.get_recent_prioritization_fees(&[]).await.unwrap();
+
+    if prioritization_fees.is_empty() || matches!(tier, PriorityFeeTier::None) {
+        return Ok(None);
+    }
+    let mut non_zero_fees: Vec<u64> = prioritization_fees
+        .iter()
+        .map(|fee| fee.prioritization_fee)
+        .filter(|&fee| fee > 0) // Keep only non-zero fees
+        .collect();
+    if non_zero_fees.is_empty() {
+        return Ok(Some(0));
+    }
+    non_zero_fees.sort_unstable();
```

Comment:
Why sort unstable?

Reply:
Fromt the docs I understand that `sort_unstable()` is usually faster than `sort()`, because it doesn't try to preserve the order of equal elements. Since we're dealing with u64s, I think that's fine. Also, it requires less memory and performs the sort in-place.

Reply:
Fair enough. I believe normal `sort` is also in-place tho